src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -49,6 +52,9 @@ class ProxyService(proxyConfig: ProxyConfig, dbRef: DbReference, clusterDnsCache
       case None =>
         // handled upstream by the LeoRoutes ExceptionHandler
         throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Could not find notebook server for $googleProject/$clusterName"))
```

Comment:
Will switch to a LeoException.

It could mean that the cluster does not exist, or it exists but it hasn't initialized (doesn't have an IP) yet. In that case the user just has to wait and try again. Assuming the "get cluster details" will aid in that.

Reply:
i think those two cases should behave differently to the user. the first can reasonably say "i don't know what you're talking about" but the second seems like a legitimate use of HTTP 420 to me: chill out, it's not ready yet

Reply:
Sure, I think we can distinguish between the 2 in code